apps/common/components/Table.tsx

```diff
@@ -0,0 +1,97 @@
+import {useCallback, useMemo, useState} from 'react';
+import {sort} from '@veYFI/utils';
+import IconChevronPlain from '@common/icons/IconChevronPlain';
+
+import type {ReactElement} from 'react';
+
+type TSortOrder = 'asc' | 'desc' | '';
+
+type TState<T> = {
+	sortedBy: Extract<keyof T, string> | undefined, 
+	order: TSortOrder
+}
+
+const switchOrder = (order: TSortOrder): TSortOrder => order === '' ? 'desc' : order === 'desc' ? 'asc' : '';
+
+type TMetadata<T> = {
+	key: Extract<keyof T, string>;
+	label: string;
+	className?: string;
+	sortable?: boolean;
+	fullWidth?: boolean;
+	columnSpan?: number;
+	format?: (item: T) => string;
+	transform?: (item: T) => ReactElement;
+}
+  
+type TTableProps<T> = {
+	metadata: TMetadata<T>[];
+	data: T[];
+	columns?: number;
+	initialSortBy?: Extract<keyof T, string>;
+	onRowClick?: (item: T) => void;
+}
+
+function Table<T>({metadata, data, columns, initialSortBy, onRowClick}: TTableProps<T>): ReactElement {
```

Comment:
Table looks fine 👍 I agree that these generic components should be the way forward